src/main/java/roomescape/reservation/service/ReservationService.java

```diff
@@ -0,0 +1,251 @@
+import roomescape.reservation.time.domain.ReservationTime;
+import roomescape.reservation.time.domain.ReservationTimeId;
+import roomescape.theme.domain.Theme;
+import roomescape.theme.domain.ThemeId;
+import roomescape.reservation.waiting.domain.Waiting;
+import roomescape.reservation.waiting.domain.WaitingWithRank;
+import roomescape.reservation.dto.request.FilteringReservationRequest;
+import roomescape.reservation.dto.request.ReservationCreateRequest;
+import roomescape.reservation.dto.response.BookedReservationTimeResponse;
+import roomescape.reservation.dto.response.MyReservationsResponse;
+import roomescape.reservation.dto.response.ReservationResponse;
+import roomescape.reservation.dto.response.ReservationTimeResponse;
+import roomescape.reservation.payment.dto.request.PaymentRequest;
+import roomescape.reservation.payment.service.PaymentService;
+import roomescape.reservation.repository.ReservationRepository;
+import roomescape.reservation.time.repository.ReservationTimeRepository;
+import roomescape.theme.repository.ThemeRepository;
+import roomescape.reservation.waiting.repository.WaitingRepository;
+
+@Service
+public class ReservationService {
+
+    private final PaymentService paymentService;
+    private final ReservationRepository reservationRepository;
+    private final ReservationTimeRepository reservationTimeRepository;
+    private final ThemeRepository themeRepository;
+    private final MemberRepository memberRepository;
+    private final WaitingRepository waitingRepository;
+
+    public ReservationService(
+            final PaymentService paymentService,
+            final ReservationRepository reservationRepository,
+            final ReservationTimeRepository reservationTimeRepository,
+            final ThemeRepository themeRepository,
+            final MemberRepository memberRepository,
+            final WaitingRepository waitingRepository
+    ) {
+        this.paymentService = paymentService;
+        this.reservationRepository = reservationRepository;
+        this.reservationTimeRepository = reservationTimeRepository;
+        this.themeRepository = themeRepository;
+        this.memberRepository = memberRepository;
+        this.waitingRepository = waitingRepository;
+    }
+
+    public List<ReservationResponse> getAll() {
+        List<Reservation> reservations = reservationRepository.findAll();
+
+        return reservations.stream()
+                .map(ReservationResponse::from)
+                .toList();
+    }
+
+    public List<MyReservationsResponse> getAllMyReservations(final LoginMember loginMember) {
+        List<Reservation> reservations = reservationRepository.findAllByMemberId(new MemberId(loginMember.id()))
+                .stream()
+                .toList();
+        List<WaitingWithRank> waitingWithRanks = waitingRepository.findAllWaitingWithRankByMemberId(
+                        new MemberId(loginMember.id()))
+                .stream()
+                .toList();
```

Comment:
`stream().toList()` 생략해도 동일합니다.

```suggestion
        List<Reservation> reservations = reservationRepository.findAllByMemberId(new MemberId(loginMember.id()));
        List<WaitingWithRank> waitingWithRanks = waitingRepository.findAllWaitingWithRankByMemberId(
                        new MemberId(loginMember.id()));
```

Reply:
아마도 이전 미션에서`Reservation`만 조회해서 반환하는 요구사항이어서 바로 응답으로 변환해 반환했기 때문에 그 때의 코드가 이어진것 같아요. 제대로 확인을 못했었네요

```java
reservationRepository.findAllByMemberId(id)
        .stream()
        .map(MyReservationsResponse::from)
        .toList();
```

위 코드를 map()만 지워서 그대로 사용한 것 같네요